lib/modules/datasource/deb/index.ts

```diff
@@ -0,0 +1,396 @@
+          compression,
+          compressedFile,
+          lastTimestamp,
+        );
+      } catch (error) {
+        logger.error(
+          {
+            compression,
+            error: error.message,
+          },
+          `Failed to download package file from ${componentUrl}`,
+        );
+
+        continue;
+      }
+
+      if (wasUpdated || !lastTimestamp) {
+        try {
+          await DebDatasource.extract(
+            compressedFile,
+            compression,
+            extractedFile,
+          );
+          lastTimestamp = await this.getFileCreationTime(extractedFile);
+        } catch (error) {
+          logger.error(
+            {
+              componentUrl,
+              compression,
+              error: error.message,
+            },
+            `Failed to extract package file from ${compressedFile}`,
+          );
+        } finally {
+          await fs.rmCache(compressedFile);
+        }
+      }
+
+      if (!lastTimestamp) {
+        //extracting went wrong
+        continue;
+      }
+
+      return { extractedFile, lastTimestamp };
+    }
+
+    throw new Error(`No compression standard worked for ${componentUrl}`);
+  }
+
+  /**
+   * Downloads a package file if it has been modified since the last download timestamp.
+   *
+   * @param basePackageUrl - The base URL of the package.
+   * @param compression - The compression method used (e.g., 'gz').
+   * @param compressedFile - The path where the compressed file will be saved.
+   * @param lastDownloadTimestamp - The timestamp of the last download.
+   * @returns True if the file was downloaded, otherwise false.
+   */
+  async downloadPackageFile(
+    basePackageUrl: string,
+    compression: string,
+    compressedFile: string,
+    lastDownloadTimestamp?: Date,
+  ): Promise<boolean> {
+    const baseReleaseUrl = getBaseReleaseUrl(basePackageUrl);
+    const packageUrl = `${basePackageUrl}/Packages.${compression}`;
+    let needsToDownload = true;
+
+    if (lastDownloadTimestamp) {
+      needsToDownload = await this.checkIfModified(
+        packageUrl,
+        lastDownloadTimestamp,
+      );
+    }
+
+    if (needsToDownload) {
+      const inReleaseContent = await this.fetchInReleaseFile(baseReleaseUrl);
+      const expectedChecksum = parseChecksumsFromInRelease(
+        inReleaseContent,
+        // path to the Package.gz file
+        packageUrl.replace(`${baseReleaseUrl}/`, ''),
+      );
+
+      const readStream = this.http.stream(packageUrl);
+      const writeStream = fs.createCacheWriteStream(compressedFile);
+      await fs.pipeline(readStream, writeStream);
+      logger.debug(
+        { url: packageUrl, targetFile: compressedFile },
+        'Downloading Debian package file',
+      );
+
+      const actualChecksum = await computeFileChecksum(compressedFile);
+
+      if (actualChecksum !== expectedChecksum) {
+        await fs.rmCache(compressedFile);
+        throw new Error('SHA256 checksum validation failed');
+      }
```

Comment:
```suggestion
      const readStream = this.http.stream(packageUrl);
      const writeStream = fs.createCacheWriteStream(compressedFile);
      await fs.pipeline(readStream, writeStream);
      logger.debug(
        { url: packageUrl, targetFile: compressedFile },
        'Downloading Debian package file',
      );

      const actualChecksum = await computeFileChecksum(compressedFile);

      try {
        const inReleaseContent = await this.fetchInReleaseFile(baseReleaseUrl);
        const expectedChecksum = parseChecksumsFromInRelease(
          inReleaseContent,
          // path to the Package.gz file
          packageUrl.replace(`${baseReleaseUrl}/`, ''),
        );
        if (actualChecksum !== expectedChecksum) {
          logger.debug(
            { url: baseReleaseUrl },
            'SHA256 checksum validation failed',
          );
          return false;
        }
      } catch (error) {
        // This is expected to fail for Artifactory if GPG verification is not enabled
        logger.debug({ url: baseReleaseUrl }, 'Could not fetch InRelease file');
      }
```
I have tested against my Artifactory instance and it seems like that one has no InRelease files if GPG verification is not enabled.